sdk-core/src/test/java/care/data4life/sdk/RecordServiceCreationIntegration.kt

```diff
@@ -317,19 +315,23 @@ class RecordServiceCreationIntegration : RecordServiceIntegrationBase() {
         val resource = buildDocumentReferenceFhir3()
 
         val tags = mapOf(
-                "partner" to "partner=TEST",
-                "client" to "client=TEST",
-                "fhirversion" to "fhirversion=3.0.1",
+                "partner" to "partner=$PARTNER_ID".toLowerCase(),
+                "client" to "client=${
+                    URLEncoder.encode(CLIENT_ID.toLowerCase(), StandardCharsets.UTF_8.displayName())
+                }",
+                "fhirversion" to "fhirversion=${
+                    "3.0.1".replace(".", "%2e")
```

Comment:
This looks like a breaking change! If a FHIR version was `fhirversion=3.0.1` and is now `fhirversion=3%2e0%2e1`, we would have different tags after encryption which result in different results when querying the RecordService API. Is the JS also encoding the fhir version tag? For changing this we would need a migration mechanism, to keep existing datasets queryable.

Reply:
JS applies the encoding in this way. See [here](https://github.com/d4l-data4life/js-sdk/blob/6e043843c525da93369db7c654c8515c8af6ac89/src/lib/stringUtils.ts#L2-L7), [here](https://github.com/d4l-data4life/js-sdk/blob/6e043843c525da93369db7c654c8515c8af6ac89/src/lib/taggingUtils.ts#L70-L76) and [here](https://github.com/d4l-data4life/js-sdk/blob/6e043843c525da93369db7c654c8515c8af6ac89/src/lib/taggingUtils.ts#L54-L56) .
My guess is that it supposed to overcome some compatibility issues regarding UTF-8/UTF-16 (but I am not sure about this). The queries itself should be fine since the tags and annotations are encoded and encrypted the very same way. (See [here](https://github.com/d4l-data4life/hc-sdk-kmp/blob/1b22b53622f569c70f97f70e4c32c9e5f4e8910e/sdk-core/src/main/java/care/data4life/sdk/RecordService.kt#L274-L280)).
However what I do not know (yet) is what the backend is doing, but since JS does it this way, I think we should be okish (but this actually my point of uncertainty).

Reply:
The backend does nothing beside searching for exactly the representation of a tag the SDK is generating. 

New behaviour: `fhirversion=3.0.1` -> encode -> `fhirversion=3%2e0%2e1` -> encrypt -> `abzrAn37nmdnud8u=` -> base64 -> `Y2xpZW50PWI0NiUyM3Rlc3Q`

The old behaviour: `fhirversion=3.0.1` -> encrypt -> `cGFydG5lcj1iNDY=` -> base64 -> `ZmhpcnZlcnNpb249My=`

`Y2xpZW50PWI0NiUyM3Rlc3Q` is the identifier the backend is searching for and is not the same as `ZmhpcnZlcnNpb249My=`

The old behaviour is actually a bug as it should have been encoded too. But now we have Java/Android that generate `fhirversion=3.0.1` = `ZmhpcnZlcnNpb249My=` and iOS/JS where `fhirversion=3.0.1` = `Y2xpZW50PWI0NiUyM3Rlc3Q`

So we need to have a migration or compatibility in place to fix this bug.

Reply:
As discussed: Agree!